src/GraphQL.Tests/Bugs/Bug1699InvalidEnum.cs

```diff
@@ -63,10 +63,10 @@ private void AssertQueryWithError(string query, string result, string message, i
         public void Input_Enum_Valid_Variable() => AssertQuerySuccess("query($arg: Bug1699Enum!) { input(arg: $arg) }", @"{ ""input"": ""Grumpy"" }", "{\"arg\":\"GRUMPY\"}".ToInputs());
 
         [Fact]
-        public void Input_Enum_InvalidEnum_Variable() => AssertQueryWithError(@"query($arg: Bug1699Enum!) { input(arg: $arg) }", null, "Variable \u0027$arg\u0027 is invalid. Unable to convert \u0027DOPEY\u0027 to \u0027Bug1699Enum\u0027", 0, 0, (object[])null, code: "INVALID_VALUE", inputs: "{\"arg\":\"DOPEY\"}");
```

Comment:
I have a question why we are losing the error code. I can see that we are taking it from a inner exception. But what if there is no inner exception (as in your fixed version)? Is it worth setting the error code anyway, and what value will it have? And after that I thought about why we even do such checks at the execution stage. This must be validated by the validation rules.

Then I checked `ArgumentsOfCorrectType` and see this inside `IsValidLiteralValue`:
![изображение](https://user-images.githubusercontent.com/21261007/89648284-aa91e100-d8c7-11ea-8c78-373c211ea789.png)

I think it's worth working on it.

Reply:
Technically the code is all performed correctly right now, even if not clear.  Variables are all validated within `GetVariableValues` prior to execution, returning an ExecutionError with the proper location information -- just not within a validation rule.  (This PR patches the exception handling code so the location is returned as intended.)  The above code is also correct because `GetVariableValues` will always validate the value types, so there is no need for redundant code to run within `IsValidLiteralValue`.

Perhaps it would be better if there was another validation rule such as `ValidateVariableValues`.  But this already occurs within `GetVariableValues`, and any additional validation rule would be redundant (since `GetVariableValues` still needs to deserialize the values prior to exeuction).